### PR TITLE
83952 Update signature block behavior - form 10-7959c

### PR DESCRIPTION
**Pull request by michaelclement**
## Summary

This PR adds a new `certifierRole` form page and a customized signature/statement of truth block. The new signature block allows us to either verify that a beneficiary has typed their own name, or allow an arbitrary name to be typed if the user has specified they are a third party.

- Added screen to collect what type of signature we need
- Added custom signature block that checks either beneficiary name or collects arbitrary string
- Updated unit tests
- Updated end to end tests
- Team: IVC CHAMPVA
- Flipper: NA

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/83952

## Testing done

- Manual
- Unit
- E2E

## Screenshots

|         | Third Party | Beneficiary |
| ------- | ------ | ----- |
| Signature type screen  |        |Screenshot 2024-06-13 at 17 51 03|
| Signature block |Screenshot 2024-06-13 at 17 08 44|Screenshot 2024-06-13 at 17 08 40|

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

NA

### CODE REVIEW

File: src/applications/ivc-champva/10-7959C/chapters/formSignature.js
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/chapters/formSignature.js:1]: Adds a page with radio buttons to collect whether or not the signer is the beneficiary or a third party.

File: src/applications/ivc-champva/10-7959C/components/CustomAttestation.jsx
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/components/CustomAttestation.jsx:1]: Comments pretty much explain it, but this is the custom signature + agreement checkbox. Most noteworthy is the fact that we either check signature against beneficiary name, or we just accept a plain string.

File: src/applications/ivc-champva/10-7959C/config/form.js
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/config/form.js:1]: Replace existing statement of truth with new custom attestation component, and add signature type collection screen to bottom.

File: src/applications/ivc-champva/10-7959C/tests/10-7959C.cypress.spec.js
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/tests/10-7959C.cypress.spec.js:1]: Updated so we can type in the new signature block and check "i agree"

File: src/applications/ivc-champva/10-7959C/tests/config/form.unit.spec.jsx
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/tests/config/form.unit.spec.jsx:1]: This test no longer applies

File: src/applications/ivc-champva/10-7959C/tests/e2e/fixtures/data/maximal-test.json
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/tests/e2e/fixtures/data/maximal-test.json:1]: Minor adjustment to E2E test data in all 10-7959c E2E jsons - updated so we have the right data to interact with the signature radio buttons + new signature field.

**Review by reiting — APPROVED** [review]

**Review by cloudmagic80 — APPROVED** [review]

**Review by CBonade — APPROVED** [review]

**Pull request merged**